src/Redmine/Api/AbstractApi.php

```diff
@@ -28,9 +28,14 @@ public function __construct(Client $client)
      * Returns whether or not the last api call failed.
      *
      * @return bool
+     *
+     * @deprecated This method does not correctly handle 2xx codes that are not 200 or 201, use \Redmine\Client\Client::getLastResponseStatusCode() instead
+     * @see Client::getLastResponseStatusCode() for checking the status code directly
      */
     public function lastCallFailed()
     {
+        @trigger_error('The '.__METHOD__.' method is deprecated, use '.Client::class.'::getLastResponseStatusCode() instead.', E_USER_DEPRECATED);
```

Comment:
Using `::class` will give you the FQCN without a leading backslash. I'm not sure, but this could confuse some IDEs?
My recommendation would simply writing the classname, like in the `@deprecated` phpdoc.
```diff
- ... use '.Client::class.'::getLastResponseStatusCode() instead.', E_USER_DEPRECATED);
+ ... use \Redmine\Client\Client::getLastResponseStatusCode() instead.', E_USER_DEPRECATED);
```